QEC_BitFlipCode/Tests.qs

```diff
@@ -91,14 +92,20 @@ namespace Quantum.Kata.QEC_BitFlipCode {
         using (register = Qubit[3]) {
             // prepare basis state to test on
             statePrep(register);
+
+            ResetOracleCallsCount();
+
             let res = MeasureParity(register);
             
             // check that the returned parity is correct
-            EqualityFactB(res == Zero, parity == 0, $"Failed on {stateStr}.");
+            Fact((res == Zero) == (parity == 0), $"Failed on {stateStr}.");
             
             // check that the state has not been modified
             Adjoint statePrep(register);
             AssertAllZero(register);
+
+            let nm = GetOracleCallsCount(M) + GetOracleCallsCount(Measure);
```

Comment:
That's odd, does CounterSimulator not catch calls from within `M`? In [`Intrinsic.qs`](https://github.com/microsoft/qsharp-runtime/blob/master/src/Simulation/Intrinsic/Intrinsic.qs#L724), `M` is implemented by a call to `Measure`, such that I'd naïvely thought this would double-count.

Reply:
I just double-checked that one call to `M` yields 1 count of `M` and 0 counts of `Measure`, so something is working differently than you expect. `MResetZ` also counts as 1 `M` and 0 `Measure`.

Reply:
CounterSimulator derives from Quantum Simulator, which provides an intrinsic implementation of `M` that overrides the implementation in `Instrinsic.qs`.

Reply:
Thank you for the explanation, this inconsistency was worrying me :-)

Reply:
Ah, of course. I forgot that `QuantumSimulator` has an implementation for both `M` and `Measure`, sorry. Thanks!